File: src/main/java/model/json/adapter/PosixDateAdapter.java
package model.json.adapter;

import com.squareup.moshi.FromJson;
import com.squareup.moshi.ToJson;

import java.util.Date;

public class PosixDateAdapter {

    public @FromJson @PosixDate Date fromJson(Long date) {
        return new Date(date);
    }

    public @ToJson Long toJson(@PosixDate Date date) {
        return date.getTime();
    }

}

File: src/main/java/model/json/adapter/PosixDateAdapter.kt
package model.json.adapter

import com.squareup.moshi.FromJson
import com.squareup.moshi.ToJson

import java.util.Date

class PosixDateAdapter {

    @FromJson
    @PosixDate
    fun fromJson(date: Long): Date = Date(date)


    @ToJson
    fun toJson(@PosixDate date: Date): Long = date.time


}
